pkg/util/wait/wait_test.go

```diff
@@ -123,10 +123,11 @@ func TestPollImmediate(t *testing.T) {
 		invocations++
 		return true, nil
 	})
-	fp := fakePoller{max: 0, wg: sync.WaitGroup{}}
+	fp := fakePoller{max: 0}
 	if err := pollImmediateInternal(fp.GetWaitFunc(), f); err != nil {
 		t.Fatalf("unexpected error %v", err)
 	}
+	// We don't need to wait for fp.wg, as pollImmediate shouldn't call WaitFunc at all.
```

Comment:
If `pollImmediateInternal` is not supposed to call WaitFunc then why we pass it there?